lib/phoenix_live_view/upload_channel.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Phoenix.LiveView.UploadChannel do
+  @moduledoc false
+  use Phoenix.Channel, log_handle_in: false
+
+  require Logger
+
+  alias Phoenix.LiveView.{Static, Channel}
+
+  def cancel(pid) do
+    GenServer.call(pid, :cancel)
+  end
+
+  def consume(pid, entry, func) when is_function(func, 1) or is_function(func, 2) do
+    GenServer.call(pid, {:consume, entry, func})
+  end
+
+  @impl true
+  def join(_topic, auth_payload, socket) do
+    %{"token" => token} = auth_payload
+
+    with {:ok, %{pid: pid, ref: ref}} <- Static.verify_token(socket.endpoint, token),
+         {:ok, config} <- Channel.register_upload(pid, ref),
+         %{max_file_size: max_file_size, chunk_timeout: chunk_timeout} = config,
+         {:ok, path} <- Plug.Upload.random_file("live_view_upload"),
```

Comment:
Since `Plug` is a transitive dependency of `phoenix`, should we add it to mix.exs?

Reply:
I would not worry about it. If we add it with ~>, then it is something more to update later on. If we add with >= 0.0.0, then it doesn't really matter. I think it is fine to assume phoenix is always with plug.